### ISSUE TITLE
Inject HttpRequest

### ISSUE THREAD

**Issue by tomfaltesek** [BorisWilhelms/azure-function-dependency-injection#32]
I have services that depend on `IHttpContextAccessor` so that certain HTTP Headers and claim data can be accessed. Is there any way to configure DI for some form of the current HttpRequest?

**Comment by ranouf** [comment]
What I use is:
```
        public IServiceProvider Build()
        {
            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
        }
```

But currently, Startup seems to be not invoked with .net core 2.2 ...

**Comment by tomfaltesek** [comment]
> What I use is:
> 
> ```
>         public IServiceProvider Build()
>         {
>             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
>         }
> ```
> But currently, Startup seems to be not invoked with .net core 2.2 ...

Yeah, that works great for a regular dotnet core web project. It does not work for Azure Functions. The `HttpContext` property is always null. Perhaps I'm doing something incorrectly.

**Comment by DonFrazier-zz** [comment]
Did you include an [assembly: FunctionsStartup(typeof(MyStartupClass))] attribute somewhere in your function?

**Comment by nickwinger** [comment]
i also would like to be able to inject the httpRequest into a service, e.g. token validation etc.

**Comment by firstdowns** [comment]
This is pretty old, but I needed to do this as well.  With the newer injection system, in your Startup class, you can use:

`public override void Configure(IFunctionsHostBuilder builder)
{
     builder.Services.AddHttpContextAccessor();
}`

Then simply inject IHttpContextAccessor where you need it.    Seems to work just fine.

**Comment by nickwinger** [comment]
Hi,

i cannot see your post…
Which version of which library is now supporting this ?

//nick
